iwyu_ast_util.cc

```diff
@@ -1183,6 +1185,17 @@ bool IsBuiltinFunction(const clang::NamedDecl* decl) {
   return false;
 }
 
+bool IsImplicitlyInstantiated(const clang::FunctionDecl* decl) {
```

Comment:
Maybe `IsImplicitlyInstantiatedDefn`? Or maybe only definitions can be implicitly instantiated anyway?

Reply:
No, clang marks as implicit instantiations also references to specializations of templates without definitions. Honestly, I'm still not sure whether it is valid C++ or not... But implementations support that.